### ISSUE TITLE
Refactor tree in state and NotesMindMapReducer.js

### ISSUE THREAD

**Issue by DarthVanger** [DarthVanger/NeuralNotes#63]
## TL;DR
Store the notes tree as `nodes` and `edges` arrays, instead of storing it as a tree, as it is stored 
now.

## Explanation
Currently the notes tree is stored as
```
rootNote: {
  children: [
     { id: '127ae9f02', name: 'Dreams', children: [ ... ] },
     { id: 'se92s2f02', name: 'Business', children: [ ... ] },
   ...
  ]
}
```
This data structure is hard to clone. There is a function in reducer to change `rootNote` is the state, every time a note changes: https://github.com/DarthVanger/NeuralNotes/blob/master/src/components/NotesMindMap/NotesMindMapReducer.js#L133

The `NotesMindMapComponent` renders the tree, starting from `rootNote`: https://github.com/DarthVanger/NeuralNotes/blob/master/src/components/NotesMindMap/NotesMindMapComponent.js#L150

So to re-render the tree, we have to update the rootNote.

`NotesMindMapComponent` transforms the tree into 2 arrays: `nodes` and `edges`, because it is the format accepted by the `vis-network.js`.

I think we should store the tree in this format in the state, to make life easier: to have 2 arrays instead of a tree. This way it should be easier to work with the notes in reducers, and re-render only the nodes that changed, not the whole tree.

Moreover storing it as `nodes` and `edges` arrays allows having multiple trees. It might be useful for example when we do search: we can render the found node in a separate tree.
